homeassistant/components/airzone/climate.py

```diff
@@ -56,28 +46,28 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-HVAC_ACTION_LIB_TO_HASS: Final[dict[OperationMode, str]] = {
-    OperationMode.STOP: CURRENT_HVAC_OFF,
-    OperationMode.COOLING: CURRENT_HVAC_COOL,
-    OperationMode.HEATING: CURRENT_HVAC_HEAT,
-    OperationMode.FAN: CURRENT_HVAC_FAN,
-    OperationMode.DRY: CURRENT_HVAC_DRY,
+HVAC_ACTION_LIB_TO_HASS: Final[dict[OperationMode, HVACAction]] = {
+    OperationMode.STOP: HVACAction.OFF,
+    OperationMode.COOLING: HVACAction.COOLING,
+    OperationMode.HEATING: HVACAction.HEATING,
+    OperationMode.FAN: HVACAction.FAN,
+    OperationMode.DRY: HVACAction.DRYING,
 }
-HVAC_MODE_LIB_TO_HASS: Final[dict[OperationMode, str]] = {
-    OperationMode.STOP: HVAC_MODE_OFF,
-    OperationMode.COOLING: HVAC_MODE_COOL,
-    OperationMode.HEATING: HVAC_MODE_HEAT,
-    OperationMode.FAN: HVAC_MODE_FAN_ONLY,
-    OperationMode.DRY: HVAC_MODE_DRY,
-    OperationMode.AUTO: HVAC_MODE_HEAT_COOL,
+HVAC_MODE_LIB_TO_HASS: Final[dict[OperationMode, HVACMode]] = {
+    OperationMode.STOP: HVACMode.OFF,
+    OperationMode.COOLING: HVACMode.COOL,
+    OperationMode.HEATING: HVACMode.HEAT,
+    OperationMode.FAN: HVACMode.FAN_ONLY,
+    OperationMode.DRY: HVACMode.DRY,
+    OperationMode.AUTO: HVACMode.HEAT_COOL,
 }
 HVAC_MODE_HASS_TO_LIB: Final[dict[str, OperationMode]] = {
```

Comment:
```suggestion
HVAC_MODE_HASS_TO_LIB: Final[dict[HVACMode, OperationMode]] = {
```

Reply:
Due to inheritance issues in the method where it is used, I have kept `HVACMode | str`